### ISSUE TITLE
Mapping of displacement fields

### ISSUE THREAD

**Issue by jbanusco** [nipy/nitransforms#185]
Hello, I believe there is a bug during the mapping of deformation fields. 

In the .map method of the DenseFieldTransform class, I believe the locations that are mapped out-of-domain should be set to the initial coordinates instead of 0. Otherwise, if for example you want to go back to the displacement field by subtracting the reference coordinates you end-up with very large displacements instead of 0. 

```
def map(self, x, inverse=False):
        ....
        ....
        new_map = np.vstack(tuple(
            map_coordinates(
                self._field[..., i],
                ijk.T,
                order=1,
                mode="constant",
                cval=0,
                prefilter=False,
            ) for i in range(self.reference.ndim)
        )).T

        # Now, where is 0 i set the original coordinates value -- no transformation/displacement
        new_map[np.isclose(new_map, 0)] = x[np.isclose(new_map, 0)]
        return new_map
```

Thanks for your work!

**Comment by effigies** [comment]
Yes, I think this is correct. I wonder if a more direct approach would be:

```diff
         return np.vstack(tuple(
-            map_coordinates(
-                self._field[..., i],
+            ijk.T + map_coordinates(
+                self._deltas[..., i],
                 ijk.T,
                 order=3,
                 mode="constant",
                 cval=0,
                 prefilter=True,
             ) for i in range(self.reference.ndim)
         )).T
```

By interpolating the deltas, we avoid needing a second step. We could also separate displacement fields from deformation fields, and treat displacement fields as I do and deformation fields as you do, which would prevent adding unnecessary floating point error to the mix.

**Comment by effigies** [comment]
> Deformation fields (SPM's preferred option) are more accurate than displacements fields (ANTs, FSL).

Is this still true when you start by converting from deltas to deformations? If you start with a deformation, then yes, you get:

```
interpolate(deformation, ijk)
# vs
ijk + interpolate(deformation - ijk, ijk)
```

If you start with deltas, then you get

```
interpolate(ijk + deltas, ijk)
# vs
ijk + interpolate(deltas, ijk)
```

I'm not sure why doing the addition in one place vs the other would accumulate more floating point error.

**Comment by oesteban** [comment]
> Is this still true when you start by converting from deltas to deformations?

Of course it's not, but you start with what you want. In general, I think converting to deformation at an early stage is the best way of keeping both implementations close to one another. I'm happy to revisit this though, if my viewpoint is too simplistic, please open an issue in that case.

Regarding this particular issue, the bug pointed out by @jbanusco was closed by his contribution in #188 so I'll close this one. There's #202 open to follow up on that one with a necessary test. Therefore, considering those two and the ancestral issue #76. I'm marking this one as done. Thanks again, @jbanusco for your PR!
